File: randomForestRegression.py
# -*- coding: utf-8 -*-
"""
Created on Wed Dec 25 19:56:21 2019

@author: 138709
"""

# Random Forest Regression

# Import the libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

# Import the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:,1].values
X = X.reshape(-1,1)
Y = dataset.iloc[:,2].values

# Fit the Random Forest Regression Model
from sklearn.ensemble import RandomForestRegressor
model = RandomForestRegressor(n_estimators = 300, random_state = 0)
model.fit(X,Y)


# Predicting the result
YPred = model.predict(6.5)


# Visualising the Random Forest Regression with higher dimensions
XGrid = np.arange(min(X), max(X), 0.01)
XGrid = XGrid.reshape((len(XGrid),1))
plt.scatter(X,Y, color = 'red')
plt.plot(XGrid, model.predict(XGrid), color = 'blue')
plt.title('Random Forest Regression')
plt.xlabel('Position')
plt.ylabel('Salaries')
plt.show()

File: randomForestRegression.R
# Random Forest Regression

# Install the packages
install.packages("randomForest")
library(randomForest)
install.packages("ggplot2")
library(ggplot2)

# Import the dataset
dataset <- read.csv('Position_Salaries.csv')


# Fit the Random Forest Model
set.seed(1234)
# model <- randomForest(Salary ~ Level, data = dataset, ntree = 10)
model <- randomForest(x = dataset[2], y = dataset$Salary, ntree = 100)


# Predicting the results
YPred <- predict(model, data.frame(Level = 6.5))


# Visualise the result in higher dimensions
xGrid <- seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() + geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'red') + geom_line(aes(x = xGrid, y = predict(model, newdata = data.frame(Level = xGrid))), colour = 'blue') + ggtitle('Random Forest Visualisation') + xlab('Level') + ylab('Salary')
